From 0a5c9fe4e62bd76edc346bdf07439058d025990e Mon Sep 17 00:00:00 2001
From: John Swinbank <swinbank@transientskp.org>
Date: Thu, 17 May 2012 12:01:36 +0200
Subject: [PATCH] Add ability to set niceness & IO priority.

Adds two new optional entries to the [cluster] section of pipeline.cfg which
change the scheduling of remote processes.

niceness - Adjust niceness.
iopriority - Set IO priority (1-3; 3 is lowest; 1 is not available for
             non-root).
---
 .../framework/lofarpipe/support/remotecommand.py   |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/CEP/Pipeline/framework/lofarpipe/support/remotecommand.py b/CEP/Pipeline/framework/lofarpipe/support/remotecommand.py
index c468d54..4d480fd 100644
--- a/CEP/Pipeline/framework/lofarpipe/support/remotecommand.py
+++ b/CEP/Pipeline/framework/lofarpipe/support/remotecommand.py
@@ -8,6 +8,7 @@
 from __future__ import with_statement
 from collections import defaultdict
 from threading import BoundedSemaphore
+from ConfigParser import Error as ConfigParserError
 
 import re
 import os
@@ -71,9 +72,23 @@ def run_remote_command(config, logger, host, command, env, arguments = None):
     """
     try:
         method = config.get('remote', 'method')
-    except:
+    except ConfigParserError:
         method = None
 
+    try:
+        niceness = int(config.get('cluster', 'niceness'))
+        logger.debug("Using niceness %d" % niceness)
+        command = "nice --adjustment=%d %s" % (niceness, command)
+    except ConfigParserError:
+        pass
+
+    try:
+        iopriority = int(config.get('cluster', 'iopriority'))
+        logger.debug("Using IO priority %d" % iopriority)
+        command = "ionice -c %d %s" % (iopriority, command)
+    except ConfigParserError:
+        pass
+
     if method == "paramiko":
         try:
             key_filename = config.get('remote', 'key_filename')
-- 
1.7.0.4

